### PR TITLE
Support sorting buckets in storage explorer

### PR DESCRIPTION
**Pull request by joshenlim**
Support searching + sorting buckets in storage explorer, similar UX to table editor

image

### CODE REVIEW

File: apps/studio/components/layouts/StorageLayout/StorageMenu.tsx [hunk 1]
```diff
@@ -61,6 +78,34 @@ const StorageMenu = () => {
...
           </ButtonTooltip>
+
+          <InnerSideBarFilters className="px-0">
+            <InnerSideBarFilterSearchInput
+              name="search-buckets"
+              aria-labelledby="Search buckets"
+              placeholder="Search buckets..."
+              value={searchText}
+              onChange={(e) => {
+                setSearchText(e.target.value.trim())
```
Comment (alaister) [inline apps/studio/components/layouts/StorageLayout/StorageMenu.tsx:89]: Maybe move the `.trim()` inside the `.sort()` call? Doing it here breaks the spacebar when typing into the input lol

File: apps/studio/components/layouts/StorageLayout/StorageMenu.tsx [hunk 2]
```diff
@@ -39,12 +52,16 @@ const StorageMenu = () => {
 
   const { data, error, isLoading, isError, isSuccess } = useBucketsQuery({ projectRef: ref })
   const buckets = data ?? []
+  const sortedBuckets =
+    sort === 'alphabetical'
+      ? buckets.sort((a, b) => a.name.localeCompare(b.name))
```
Comment (alaister) [inline apps/studio/components/layouts/StorageLayout/StorageMenu.tsx:57]: related to previous comment, something like:
```suggestion
      ? buckets.sort((a, b) => a.name.toLowerCase().trim().localeCompare(b.name.toLowerCase().trim()))
```
wdyt?

File: apps/studio/components/layouts/StorageLayout/StorageMenu.tsx (hunk 1 above)
Comment (joshenlim) [reply apps/studio/components/layouts/StorageLayout/StorageMenu.tsx:89]: oh yes! good catch

File: apps/studio/components/layouts/StorageLayout/StorageMenu.tsx (hunk 2 above)
Comment (joshenlim) [reply apps/studio/components/layouts/StorageLayout/StorageMenu.tsx:57]: okay to do this - but does doing this cover any edge case that the original implementation wouldn't?

Comment (alaister) [reply apps/studio/components/layouts/StorageLayout/StorageMenu.tsx:57]: Should be pretty similar

Comment (joshenlim) [reply apps/studio/components/layouts/StorageLayout/StorageMenu.tsx]: okay gotcha! will just update it :)

**Review by alaister — APPROVED** [review]
LGTM!

- Tested on preview

**Pull request merged**